### DISCUSSION TITLE
DataModule property `train_transforms` was deprecated

### DISCUSSION THREAD

**Discussion by Wheest** [Lightning-AI/pytorch-lightning#12952] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I was following the [official example on running CIFAR10](https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/cifar10-baseline.html).  However, when running I get warnings deprecated features related to the transforms:

```
DataModule property `train_transforms` was deprecated in v1.5 and will be removed in v1.7.
DataModule property `val_transforms` was deprecated in v1.5 and will be removed in v1.7.
```

Given it is deprecated, what is the new way of applying data transformations to a DataModule?  What should the example be updated to use?

**Comment by rohitgr7** [comment]
It's using datamodule from pl_bolts which is quite outdated. You can still assign these transformers to any state you want and use them as per your use-cases.

**Reply by Wheest** [reply]
Thanks, however is there a recommended modern (non-deprecated) approach to achieve this, if I wanted to make my own datamodules inspired from this, or if I wanted to upstream the upgrade to pl_bolts?

**Reply by rohitgr7** [reply]
the deprecation is made for assigining train_transforms. It has now been removed on PL master so you won't even see it again after the next release. It's just that PL bolts is not maintained very well and needs a good refactor. You can still assign train_transforms as you like and ignore the warning.

```py
class LitDM(LDM):
    def __init__(...):
        super().__init__()
        self.train_transforms = ...  # this is perfectly fine
```

**Reply by Wheest** [reply]
So does that mean in the next PL release the PL bolts `CIFAR10DataModule` will stop working, since `train_transforms` is going to be removed?  Or is it just going to be a specific feature of PL Bolts going forward?

I'm keen to ensure my code will continue to work even when PL is upgraded

**Reply by akihironitta** [reply]
Hi @Wheest!

> So does that mean in the next PL release the PL bolts CIFAR10DataModule will stop working, since train_transforms is going to be removed?

No, It'll work as is. Users are still free to use the properties as they implement! It's just that those properties will be removed from the base class (`LightningDataModule`) for the reason stated in #8728.
